### PR TITLE
[Tutorials] Fix wrong vetoing of tutorials

### PR DESCRIPTION
**Pull request by hageboeck**
This depends on #5998 

- TMVA was somehow vetoing dataframe python tutorials. These have been reactivated.
- The vetoing logic in general has been overhauled, so it's more straight-forward, and the number of active / vetoed tutorials is correctly reported by cmake.
- Fix a few tutorials that wouldn't launch or compile.

### Overhaul of veto-or-fail when python packages are missing
Disable (instead of veto) python tutorials that cannot work when python packages are missing. This happens by listing the required packages when registering the test or tutorial:
```
ROOT_ADD_PYUNITTEST(pyroot_pyz_ttree_branch ttree_branch.py PYTHON_DEPS numpy)
```
Before launching the tutorial/test, cmake will run a fixture, which is a simple `import <package>`. If this passes, all dependent tutorials/tests are run. If it fails, it looks like this:
image

The advantage of this is that one doesn't even have to open the test output, because cmake already communicates clearly that the failure is due to missing packages.

As discussed in Moday's meeting, all tests with extra runtime dependencies now are also labelled `python_runtime_deps`. This gets assigned automatically if `PYTHON_DEPS` is not empty.

### CODE REVIEW

**Comment by hageboeck** [comment]
> At some point there was a discussion and I think the outcome was to always keep these tutorials in and install pandas and numba on all build nodes.

That's fair, but users who don't have either of them installed will get broken tests if they choose to run the tests.

**Comment by eguiraud** [comment]
Between silently not running tutorials (like we were doing until this PR, thank you very much for spotting and fixing it) and leaving those very rare users that run our full test suite with a few red python tutorials/tests (with error messages that say "please install pandas/numba"), I'd say the latter is preferable :smile: 

There might be a third option I don't see at the moment. In any case, I'm leaving the approval in the hands of who knows better than me :D

**Comment by pcanal** [comment]
> That's fair, but users who don't have either of them installed will get broken tests if they choose to run the tests.

Well ... for what it is worth, I am one those users ... (and  a minimal build would also be one of those users, wouldn't it be?).  If the ROOT build configure successfully and build successfully, then ctest should be clean also.   I.e. roottest failing because numba is not installed would, in my opinion, means that the cmake configure step of ROOT should fail if numba is not installed ... i.e. numba would become a hard dependency.

**Comment by Axel-Naumann** [comment]
Let's not rehash the discussion some of us already had (with the exact same arguments), and ideally not in writing on GitHub. Instead it might be more efficient to discuss this at a team meeting. I won't be able to attend on Monday; I suggest we discuss this in nine days.

**Comment by hageboeck** [comment]
> Well ... for what it is worth, I am one those users ... (and a minimal build would also be one of those users, wouldn't it be?).

I agree (hence this PR), but I think I have found a solution (@Axel-Naumann). Look at this:
```
ctest -R df026
Test project /home/shageboe/root-archNative-clang
    Start 797: tutorial-dataframe-df026_AsNumpyArrays-py
1/1 Test #797: tutorial-dataframe-df026_AsNumpyArrays-py ...***Not Run (Disabled)   0.00 sec
No tests were found!!!
```
CMake's `DISABLED` property can be used already in 3.9, and disabled tests are listed, but don't run.

File: tutorials/CMakeLists.txt [hunk 1]
```diff
@@ -251,16 +259,17 @@ if(root7)
...
-                 v7/ntuple/ntpl003_lhcbOpenData.C
-                 v7/ntuple/ntpl004_dimuon.C
-                 v7/ntuple/ntpl005_introspection.C)
-elseif(NOT dataframe)
-  set(root7_veto v7/ntuple/ntpl004_dimuon.C)
+  if(NOT dataframe)
+    set(root7_veto v7/ntuple/ntpl004_dimuon.C)
+  endif()
+else()
+  #  set(root7_veto v7/ntuple/ntpl001_staff.C
```
Comment (stwunsch) [inline tutorials/CMakeLists.txt:266]: Why do you want to keep the outcommented lines?

File: tutorials/dataframe/df007_snapshot.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 # A simple helper function to fill a test tree: this makes the example stand-alone.
 def fill_tree(treeName, fileName):
+    df = ROOT.RDataFrame(10000)
```
Comment (stwunsch) [inline tutorials/dataframe/df007_snapshot.py:14]: The code is not valid without that line, how did the tutorial run before?

Comment (hageboeck) [reply tutorials/dataframe/df007_snapshot.py:14]: It didn't run! That's why I'm making this PR!!

File: tutorials/CMakeLists.txt (hunk 1 above)
Comment (hageboeck) [reply tutorials/CMakeLists.txt:266]: That was a test. Given that the line below works, I can remove, indeed.

**Comment by hageboeck** [comment]
@phsft-bot build

**Pull request merged**